homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for Husqvarna Bluetooth integration."""
+
+from __future__ import annotations
+
+import logging
+import random
+from typing import Any
+
+from automower_ble.mower import Mower
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import BluetoothServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ADDRESS, CONF_CLIENT_ID
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_supported(discovery_info: BluetoothServiceInfo):
+    """Check if device is supported."""
+
+    _LOGGER.debug(
+        "%s manufacturer data: %s",
+        discovery_info.address,
+        discovery_info.manufacturer_data,
+    )
+
+    return any(key == 1062 for key in discovery_info.manufacturer_data)
+
+
+class HusqvarnaAutomowerBleConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Husqvarna Bluetooth."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.address: str | None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle the bluetooth discovery step."""
+
+        _LOGGER.debug("Discovered device: %s", discovery_info)
+        if not _is_supported(discovery_info):
+            return self.async_abort(reason="no_devices_found")
+
+        self.address = discovery_info.address
+        await self.async_set_unique_id(self.address)
+        self._abort_if_unique_id_configured()
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
```

Comment:
Something flaky goes on when setting up and paring the device the first time.
1)  I paired and the setup "hung" at the pairing state.
2) Reloaded the integration and it found entities but they were unavailable.
3) Reloaded again and the entities became available.

On a second try with the esphome ble gateway already "paired" I still needed to do a reload to get the entities to become available when adding the integration.

Reply:
Can you provide logs?

I found pairing the official application unreliable for the first pair as well

Reply:
> I found pairing the official application unreliable for the first pair as well

Same here